tests/unit/data/data_pipeline/test_map.py

```diff
@@ -337,3 +345,60 @@ def fn(d: int) -> int:
 
         with pytest.raises(StopIteration):
             next(iter(pipeline))
+
+    @pytest.mark.parametrize("num_parallel_calls", [1, 4, 20])
+    def test_op_saves_and_restores_its_state_non_deterministic(self, num_parallel_calls: int) -> None:  # fmt: skip
+
+        def fn(d: int) -> int:
+            return d
```

Comment:
it would be important to add a test where different tasks have different sleep(time) and to check the return order